panic when assign to nil

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"
    
func main() {
    var buf []byte
    
    buf = nil
    fmt.Println(buf)
}
```

Expected result:
```console
$ go run ./sample.go
[]
```

Got:
```console
$ yaegi ./sample.go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x17 pc=0x10a4f47]

goroutine 1 [running]:
reflect.(*rtype).Kind(...)
	/usr/local/Cellar/go/1.12.4/libexec/src/reflect/type.go:783
reflect.convertOp(0x17a0880, 0x0, 0xc000196d00)
	/usr/local/Cellar/go/1.12.4/libexec/src/reflect/value.go:2359 +0x27
reflect.Value.Convert(0x0, 0x0, 0x0, 0x1a91ba0, 0x17a0880, 0x1, 0xc00017dc10, 0xc8)
	/usr/local/Cellar/go/1.12.4/libexec/src/reflect/value.go:2349 +0x8a
github.com/containous/yaegi/interp.(*Interpreter).Cfg.func2(0xc0000e90e0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:374 +0x6261
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e90e0, 0xc00017dd50, 0xc00017dd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:104 +0xaf
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e8a50, 0xc00017dd50, 0xc00017dd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e85a0, 0xc00017dd50, 0xc00017dd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0000e80f0, 0xc00017dd50, 0xc00017dd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Cfg(0xc0000b8540, 0xc0000e80f0, 0xc0000a6680, 0x4, 0x0, 0x0, 0x4)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:24 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0000b8540, 0xc00008a240, 0x5a, 0x25a, 0xc00008a240, 0x5a, 0x0, 0xc000186b60)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:237 +0x135
main.main()
	/Users/marc/go/src/github.com/containous/yaegi/yaegi.go:59 +0x3ec
```

--- a/_test/assign1.go
+++ b/_test/assign1.go
@@ -1,0 +1,13 @@
+package main
+
+import "fmt"
+
+func main() {
+	var buf []byte
+
+	buf = nil
+	fmt.Println(buf)
+}
+
+// Output
+// []

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -367,9 +367,12 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 				src.level = level
 			case src.kind == BasicLit:
 				// TODO: perform constant folding and propagation here
-				if dest.typ.cat == InterfaceT {
+				switch {
+				case dest.typ.cat == InterfaceT:
 					src.val = reflect.ValueOf(src.val)
-				} else {
+				case src.val == nil:
+					// Assign to nil
+				default:
 					// Convert literal value to destination type
 					src.val = reflect.ValueOf(src.val).Convert(dest.typ.TypeOf())
 					src.typ = dest.typ

--- a/interp/run.go
+++ b/interp/run.go
@@ -198,6 +198,9 @@ func assign(n *Node) {
 		value1 = genValueInterface(src)
 	case dest.typ.cat == ValueT && src.typ.cat == FuncT:
 		value1 = genNodeWrapper(src)
+	case src.kind == BasicLit && src.val == nil:
+		t := dest.typ.TypeOf()
+		value1 = func(*Frame) reflect.Value { return reflect.New(t).Elem() }
 	default:
 		value1 = genValue(src)
 	}

